Add Fedora, OpenSuse and Ubuntu 16.04 packages

ellismg opened the pull request:
- Add packages for Fedora 23 and OpenSuse 13.2
 - Move the package authoring for Ubuntu into versioned folders
 - Update our selection logic for what to produce to be based on an
   actual RID instead of just a distro name, since that's now not enough
   with us building for two Ubuntu versions

ellismg commented:
@wtgodbe Can you please take a look?

/cc @gkhanna79

wtgodbe commented:
LGTM, so far as I can tell. Thanks for changing the logic in build-packages.sh as well as build.sh. Have you tested producing packages on the new distros/confirmed it's still working on the pre-existing ones?

ellismg commented:
> Have you tested producing packages on the new distros/confirmed it's still working on the pre-existing ones?

I did a build on Ubuntu 14.04 and it worked.  I'll do builds in RHEL as well.

gkhanna79 commented on build.sh:
```diff
@@ -216,25 +212,19 @@ isMSBuildOnNETCoreSupported()
-                    __isMSBuildOnNETCoreSupported=1
-                fi
-            elif [ "$__DistroName" == "rhel" ]; then
+            if [ "$__DistroRid" == "ubuntu.14.04-x64" ]; then
```
Are we not going to add 16.04 check?

ellismg replied:
I am going to change this whole thing completely soon.  Right now we don't have 16.04 CLIs.

gkhanna79 commented on src/.nuget/Microsoft.NETCore.ILAsm/fedora/23/Microsoft.NETCore.ILAsm.pkgproj:
```diff
@@ -0,0 +1,38 @@
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
+  
+  <PropertyGroup>
+    <Version>1.0.3</Version>
```
All package versions should be 1.0.2. @AlexGhiondea has a PR (https://github.com/dotnet/coreclr/pull/5216) that is doing it for existing packages, so it will be good to ensure the new definitions adhere to it as well.

ellismg replied:
Will fix, thanks for the heads up.

gkhanna79 commented:
LGTM module comments.

ellismg commented:
@gkhanna79 I addressed your feedback, can you please take another look?  I am going to finish final validation tonight on existing build targets, then I would like to merge this and bring the new distros online tomorrow.

gkhanna79 commented:
LGTM - thanks for driving this @ellismg!

gkhanna79 commented:
@ellismg Should this go to release/1.0.0 as well?

ellismg commented:
We will want to put this in before we are able to turn on the build pipeline there.  I can do this proactively later if you'd like.

gkhanna79 commented:
Thanks @ellismg Build pipeline is already enabled for release branch, so please do get this ported sooner than later.
